web/modules/custom/dept_core/dept_core.module

```diff
@@ -256,3 +258,38 @@ function dept_core_department_links(): array {
   }
   return $links;
 }
+
+/**
+ * Implements hook_form_alter().
+ */
+function dept_core_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  if ($form_id === 'views_exposed_form') {
+    if (!empty($form['dept'])) {
+      /** @var \Drupal\dept_core\DepartmentManager $dept_manager */
+      $dept_manager = \Drupal::service('department.manager');
+
+      // If the user doesn't have top level domain permissions, hide this filter option.
+      if (!\Drupal::currentUser()->hasPermission('administer domains')) {
+        /** @var \Drupal\dept_core\Entity\Department $dept */
+        $dept = $dept_manager->getCurrentDepartment();
+        $form['dept']['#value'] = $dept->id();
+        hide($form['dept']);
+        return;
+      }
+
+      // Change to select list of known depts.
+      $depts = \Drupal::service('department.manager')->getAllDepartments();
```

Comment:
You're loading this service twice, I'd replace it with $dept_manager->getAllDepartments();

Reply:
Good spot, will pop that in now!